Packing AO in the metallic roughness map

yaRnMcDonuts opened the pull request:
Updated frag shader to support reading the AmbientOcclusion value from the Red channel of the packed MetallicRoughness map. 
Adds a single boolean define that does this when set to true, so long as a metallic roughness map is also defined.

In response to this open issue: 
https://github.com/jMonkeyEngine/jmonkeyengine/issues/999

riccardobl commented on jme3-core/src/main/resources/Common/MatDefs/Light/PBRLighting.frag:
```diff
@@ -134,10 +134,12 @@ void main(){
     #endif
 
+    //ao in r channel, roughness in green channel, metallic in blue channel!
+    vec3 metallicRoughnessAoValue = vec3(1.0, 1.0, 0.0);
```
could you call it something like aoRoughnessMetallicValue to respect the order of the channels?

riccardobl commented on jme3-core/src/main/resources/Common/MatDefs/Light/PBRLighting.frag:
```diff
@@ -208,19 +210,24 @@ void main(){
-          lightMapColor = texture2D(m_LightMap, texCoord).rgb;
+        #ifdef AO_PACKED_IN_MR_MAP
+            lightMapColor = metallicRoughnessAoValue.rrr; //ao value packed in red channel of MR map 
+        #else
+            #ifdef SEPARATE_TEXCOORD    
+                lightMapColor = texture2D(m_LightMap, texCoord2).rgb;
+            #else
+                lightMapColor = texture2D(m_LightMap, texCoord).rgb;
+            #endif                                                 
        #endif
-       #ifdef AO_MAP
-         lightMapColor.gb = lightMapColor.rr;
-         ao = lightMapColor;
+       #if defined(AO_MAP) || defined(AO_PACKED_IN_MR_MAP)
+            lightMapColor.gb = lightMapColor.rr;
+            ao = lightMapColor;
        #else
-         gl_FragColor.rgb += diffuseColor.rgb * lightMapColor;
+            gl_FragColor.rgb += diffuseColor.rgb * lightMapColor;
        #endif
+       
```
I think the new code should go here:

https://github.com/jMonkeyEngine/jmonkeyengine/blob/master/jme3-core/src/main/resources/Common/MatDefs/Light/PBRLighting.frag#L227

And just replace (or multiply) the value of `vec3 ao`, so that it doesn't interfere with the lightmap code that can be used together with the ao.

riccardobl reviewed (changes requested):
Can you merge your other PR ( #1404  ) to this one?

yaRnMcDonuts commented:
I updated the name of the vec3 to aoRoughnessMetallicValue, and also switched the code to work how you suggested, so that the normal light map code is unaltered.

yaRnMcDonuts commented:
I could not find how to merge my other PR - but I managed to update this PR's branch to include the changes to PBRLighting.j3md that were in the closed pull request

riccardobl commented:
Perfect! Thank you.
